File: Selenium/Project/Activity8.py

# Import webdriver from selenium
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

# Set up the Firefox Driver with WebDriverManger
# service = FirefoxService(GeckoDriverManager().install())
driver = webdriver.Firefox()
# Start the Driver
# with webdriver.Firefox(service = service) as driver:

# Open the browser to the URL
driver.get("http://alchemy.hguy.co/crm")
wait = WebDriverWait(driver, 10)

username_input = driver.find_element(By.ID, "user_name")
username_input.send_keys("admin")

password_input = driver.find_element(By.ID, "username_password")
password_input.send_keys("pa$$w0rd")

login_button = driver.find_element(By.ID, "bigbutton")
login_button.click()

sales_link = wait.until(EC.visibility_of_element_located((By.XPATH, "//a[text()='Sales']")))
sales_link.click()

accounts_tab = driver.find_element(By.ID, "moduleTab_9_Accounts")
accounts_tab.click()

table_row = wait.until(EC.visibility_of_element_located((By.XPATH, "//tr[@class='oddListRowS1']")))
names = driver.find_elements(By.XPATH, "//tr[@class='oddListRowS1']/td[@field='name']")
for i in range(5):
    print(names[i].text)

driver.quit()

File: Selenium/Project/Activity8.java
package Project;
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

import java.time.Duration;
import java.util.List;

public class Activity8 {
    WebDriver driver;
    WebDriverWait wait;

    @BeforeClass
    public void setup() {
        WebDriverManager.firefoxdriver().setup();
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "NULL");
        driver = new FirefoxDriver();
        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        driver.get("http://alchemy.hguy.co/crm");
    }

    @Test
    public void test1(){
        driver.findElement(By.id("user_name")).sendKeys("admin");
        driver.findElement(By.id("username_password")).sendKeys("pa$$w0rd");
        driver.findElement(By.id("bigbutton")).click();
        Assert.assertTrue(driver.findElement(By.id("recentlyViewedSidebar")).isDisplayed());
        driver.findElement(By.xpath("//a[text()='Sales']")).click();
        driver.findElement(By.id("moduleTab_9_Accounts")).click();
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//tr[@class='oddListRowS1']")));
        List<WebElement> name = driver.findElements(By.xpath("//tr[@class='oddListRowS1']/td[@field='name']"));
        for(WebElement elem: name){
            System.out.println(elem.getText());
        }
    }

    @AfterClass
    public void tearDown(){
        driver.quit();
    }
}